### PR TITLE
Reduce complexity linear_discriminant_analysis.

### PR DESCRIPTION
**Pull request by NumberPiOso**
Tracking issue #2128
### **Describe your change:**
* [X] Fix a bug or typo in an existing algorithm?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [ X All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

Not really know how to create doctests for this kind of input functions.

### CODE REVIEW

**Comment by NumberPiOso** [comment]
How do I get for someone to approve this?

File: machine_learning/linear_discriminant_analysis.py [hunk 1]
```diff
@@ -245,6 +246,38 @@ def accuracy(actual_y: list, predicted_y: list) -> float:
...
+    :default: Default value in case the user does not type anything
+    :return: user's input
+    """
+    while True:
+        try:
+            user_input = input_type(input(input_msg).strip() or default)
+            if condition(user_input):
+                return user_input
+            else:
+                print(f"Your entered value is {user_input}. {err_msg}")
```
Comment (dhruvmanila) [inline machine_learning/linear_discriminant_analysis.py:275]:
```suggestion
                print(f"{user_input}: {err_msg}")
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Sorry for taking so long, just a few small changes and this is good to go.

File: machine_learning/linear_discriminant_analysis.py [hunk 2]
```diff
@@ -245,6 +246,38 @@ def accuracy(actual_y: list, predicted_y: list) -> float:
...
+                continue
+        except ValueError:
+            print("Your entered value is not numerical!")
```
[the remark this answers is not in the document]
Comment (NumberPiOso) [reply machine_learning/linear_discriminant_analysis.py:279]: Applied

**Comment by NumberPiOso** [comment]
@dhruvmanila 

I just get one check from the Travis CI system, nothing from pre-commit. 

Does this mean that it is good to go or that something is bad?

**Comment by dhruvmanila** [comment]
Your branch is based off on a very old commit, the `psf/black` automated action is deprecated. Also, do not start the workflow runs on your forked copy as some action might be automated which will make commits to your branch and that will come in this pull request as well.

Always update your fork copy to keep in sync with the main repository. If you want to make a pull request, make a branch from the latest copy and then make your changes in that branch.

File: machine_learning/linear_discriminant_analysis.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhruvmanila) [reply machine_learning/linear_discriminant_analysis.py:279]: This was not applied or a GitHub action reverted the changes.

**Comment by NumberPiOso** [comment]
Yes I get it, I suppose I always start the branches from the lastest commit.

But this pull request has been open since 19 September, almost two months, so I guess this explains why it is  200 commits behind.

If it is better to start another pull request cherry picking this changes or something let me know.

**Comment by dhruvmanila** [comment]
I think rebasing should be enough, if that won't work then try cherry-picking otherwise another PR seems like the easiest option :)

**Review by dhruvmanila — APPROVED** [review]
This seems good to go. Thank you for your contribution! And sorry for the late review.

**Comment by NumberPiOso** [comment]
Thank you for the review.

**Pull request merged**